src/components/Application.js

```diff
@@ -0,0 +1,330 @@
+                  name="major"
+                  type="text"
+                  placeholder="Computer Science"
+                  value={major}
+                  onChange={this.onChange}
+                  required
+                />
+              </Form.Group>
+		        </Col>
+		        <Col>
+		          <h5>Class Year</h5>
+			        <Form.Group>
+        				<Form.Control
+                  as="select"
+        				  name="classYear"
+        				  value={classYear}
+        				  onChange={this.onChange}
+                  required
+        				>
+                  <option value="">-</option>
+                  {years.map(year => <option key={year}>{year}</option>)}
+                </Form.Control>
+			        </Form.Group>
+		        </Col>
+		      </Row>
+		      <Row>
+		        <Col>
+		          <h5>Taken CS 112?</h5>
+              <Form.Group>
+      			    <Form.Control as="select"
+      			      name="taken112"
+      			      value={taken112}
+      			      onChange={this.onChange}
+                  required
+      			    >
+                  <option value="">-</option>
+      			      <option>Yes</option>
+      			      <option>No</option>
+      			      <option>Currently Enrolled</option>
+                </Form.Control>
+              </Form.Group>
+		        </Col>
+		        <Col>
+		          <h5>Taken CS 330?</h5>
+              <Form.Group>
+      			    <Form.Control as="select"
+      			      name="taken330"
+      			      value={taken330}
+      			      onChange={this.onChange}
+                  required
+      			    >
+      				    <option value="">-</option>
+      			      <option>Yes</option>
+      			      <option>No</option>
+      			      <option>Currently Enrolled</option>
+                </Form.Control>
+              </Form.Group>
+		        </Col>
+		      </Row>
+		      <Row>
+		        <Col>
+		          <h5>Why do you want to join UPE?</h5>
+		          <p>Around 100 Words</p>
+              <Form.Group>
+                <Form.Control
+                  name="q1"
+        				  as="textarea"
+        				  rows="7"
+        				  placeholder="..."
+        				  value={q1}
+        				  onChange={this.onChange}
+                  required
+        				/>
+              </Form.Group>
+	          </Col>
+	        </Row>
+    		  <Row>
+    		    <Col>
+    		      <h5>What do you hope to gain from joining UPE?</h5>
+    		      <p>Around 100 Words</p>
+              <Form.Group>
+                <Form.Control
+                  name="q2"
+        				  as="textarea"
+        				  rows="7"
+        				  placeholder="..."
+        				  value={q2}
+        				  onChange={this.onChange}
+                  required
+        				/>
+              </Form.Group>
+		        </Col>
+		      </Row>
+		      <Row>
+		        <Col>
+			        <h5>When are you available for interviews?</h5>
+			      </Col>
+		      </Row>
+		      <Row>
+			      <Col>
+			        <h6>Saturday - 01/01/20</h6>
```

Comment:
It might be neat to dynamically set dates in the future too.

I really like how `timeslots` works. Nice.

Reply:
Yeah I agree, once the database is setup we should make as many parts of the application as we can to be configurable.